app/models/manageiq/providers/lenovo/physical_infra_manager/event_catcher/stream.rb

```diff
@@ -27,12 +28,26 @@ def filter_fields
       { :operation => 'NOT', :field => 'eventClass', :value => '200' },
       { :operation => 'NOT', :field => 'eventClass', :value => '800' }
     ]
-    last_cn_event = get_last_cnn_from_events(@ems.id)
-    cn_operation = { :operation => 'GT', :field => 'cn', :value => last_cn_event.to_s }
-    fields.push(cn_operation) unless last_cn_event.nil?
+
+    unless @bigger_ems_ref.nil?
+      cn_operation = { :operation => 'GT', :field => 'cn', :value => @bigger_ems_ref.to_s }
+      fields.push(cn_operation)
+    end
     fields
   end
 
+  def parse_events(events)
+    events = events.sort { |x, y| Integer(x.cn) <=> Integer(y.cn) }.collect do |event|
+      ManageIQ::Providers::Lenovo::PhysicalInfraManager::EventParser.event_to_hash(event, @ems.id)
+    end
+
+    # Update the @bigger_ems_ref with the new bigger ems_ref if to exist new events
+    if events.any?
```

Comment:
Since the above loop doesn't filter out any events wouldn't this check be better up at the start of the function?
You can just `return if events.blank?`

Reply:
@agrare so, could you review again ?